src/common/date.ts

```diff
@@ -1,5 +1,6 @@
+import { utcToZonedTime } from 'date-fns-tz';
+
 export const getTodayTz = (timeZone: string) => {
   const now = new Date();
-  const timeZonedToday = now.toLocaleDateString('en', { timeZone });
-  return new Date(timeZonedToday);
+  return utcToZonedTime(now, timeZone);
```

Comment:
Since this does not really do much now if not setting hours to `0` maybe we can just inline in `src/common/users.ts`?